timemachine/cpp/src/nonbonded_all_pairs.cu

```diff
@@ -295,12 +287,14 @@ void NonbondedDense<RealType, Interpolated>::execute_device(
         gpuErrchk(cudaMemcpyAsync(
             p_ixn_count_, nblist_.get_ixn_count(), 1 * sizeof(*p_ixn_count_), cudaMemcpyDeviceToHost, stream));
 
+        // this stream needs to be synchronized so we can be sure that p_ixn_count_ is properly set.
+        cudaStreamSynchronize(stream);
+
         std::vector<double> h_box(9);
         gpuErrchk(cudaMemcpyAsync(&h_box[0], d_box, 3 * 3 * sizeof(*d_box), cudaMemcpyDeviceToHost, stream));
         // Verify that the cutoff and box size are valid together. If cutoff is greater than half the box
         // then a particle can interact with multiple periodic copies.
         const double db_cutoff = (cutoff_ + nblist_padding_) * 2;
-        cudaStreamSynchronize(stream);
```

Comment:
This sync has to be here, else the box may not be copied from the device

Reply:
I moved `cudaStreamSynchronize` back to the original location with respect to the memcpy call in 25fd4f6 (although [docs](https://docs.nvidia.com/cuda/cuda-runtime-api/api-sync-behavior.html) seem to indicate that this should only matter for copies to non-pageable host memory)